Library tags

bwlang opened the pull request:
This adds support for #tags in library datasets, auto-tagging on input (based on filenames) and copying tags to history datasets on import.

bwlang committed:
Adds tags to library dataset items, and allows users to specify that tags should be generated from filenames

bwlang committed:
shows tags in library views

bwlang committed:
According to John, this is not actually deprecated

bwlang committed:
copies tags on library items to history 
(and to copies of library items, though I don’t 
think anything called this)

dannon commented:
Nice stuff!  I'll check this out today

bwlang committed:
Removes unused import

bwlang commented:
great! - thanks @dannon.   I just committed a fix for a style violation, can you trigger the other tests (I can't easily run them locally due to tox incompatibility with conda)

bwlang committed:
Fixes style violation # comment not #comment

bwlang committed:
fixes import order to comply with flake8

bwlang commented:
I don't think you care about what I tell you ... but I'll try anyway
@galaxybot test this

nsoranzo commented:
@bwlang galaxybot only answers to people in the committers group.

jmchilton commented:
I'll work on eliminating the transient failures. It looks like some of these are legitimate failures. @bwlang do you think you are going to have time to follow up on this? If you'd like a committer to follow up and hack on this some let us know - I can't promise anything but this is a very nice start and we should try to get this in for 17.09.

bwlang commented:
@jmchilton : I have some time now... i'll at least try to suss out which tests are telling me something and which are not.

bwlang committed:
need to construct this field to avoid a conditional
check failure

bwlang commented:
@jmchilton  I think this will fix at least a few of these tests...

bwlang commented:
forced a rebase to sync up with upstream/dev... hopefully will run tests now.

bwlang committed:
Rather than trying to find every path into this 
Function, assume no tagging if dataset does not
Have a tag_using_filenames property

bwlang committed:
updates doc to include tag_using_filenames

bwlang committed:
Allow passing in tag_using_filenames via grouping

bwlang commented:
@galaxybot  test this ( I  think this test only works sometimes)

bwlang commented:
@dannon  I think the remaining failure is actually one of the tests that is not deterministic and this is ready for review.

for the failing test... Maybe it's operating on a partial download of the tar archive?

bwlang commented:
@galaxybot test this

dannon commented:
@bwlang Yep, I think you're right.  I'll go through this now, thanks for the updates!

bwlang commented:
@dannon , just rebased so this can apply to dev

dannon commented:
Had a couple of problems with this.  sa_session wasn't defined in my testing, and I'd also like to retain tags added to data libraries from histories.  I'll try to open a PR against your branch later today.  (PR merged now, the rest of the work can be future improvements)

dannon committed:
Swap two more spots initializing TagManagers to new session instead of app

bwlang committed:
Merge pull request #1 from dannon/library_tags

Swap two more spots initializing TagManagers to new session

bgruening commented:
@martenson what do you think?

martenson reviewed (approved):
This is a fairly narrow usecase but a solid groundwork for future improvements. Thank you Brad!

martenson committed:
make client

martenson committed:
Merge branch 'dev' into library_tags

martenson committed:
make client again...

nsoranzo commented on lib/galaxy/model/migrate/versions/0135_add_library_tags.py:
```diff
@@ -0,0 +1,45 @@
+
+
+LibraryDatasetDatasetAssociationTagAssociation_table = Table( "library_dataset_dataset_association_tag_association", metadata,
+                                                       Column("id", Integer, primary_key=True),
```
Indentation doesn't seem correct here.

martenson committed:
fix indentation

bgruening commented:
Double check of the migration can be found in this commit: https://github.com/bgruening/docker-galaxy-stable/commit/1eea0284ef4ec50fdebe807fa88d8898f819a39d
